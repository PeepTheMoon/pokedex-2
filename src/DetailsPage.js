import React, { Component } from 'react';
import request from 'superagent';
import PokemonItem from './PokemonItem.js';

export default class DetailsPage extends Component {
    //initializes state
    state = {
        character: {pokemon: ''}
    }
    async componentDidMount() {
        //grabs everything after the question mark in the url
        const searchParams = new URLSearchParams(window.location.search);
        //looks for pokemon key in url, sets value equal to query
        const query = searchParams.get('search');
        //gets the search query and sets the state tot he new value
        this.setState({ searchQuery: query });
        if (query) {
            //if the query has mor than one page of results
            let page = 1;
            //if the search params has a page number, gets the page number from the search params
            if (searchParams.get('page')) {
                page = searchParams.get('page');
            }
        }
        //sends the request to the api and waits for a response
        const fetchedData = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?pokemon=${this.props.match.params.name}`);
        // sets the state to the data fetched from the api to the web page
        this.setState({ character: fetchedData.body.results[0] });
    }

    render() {
        return (
            //returns the fetched pokemon data to the page in rendered html
            <div>
                <PokemonItem character={this.state.character}/>
            </div>
        )
    }
}
