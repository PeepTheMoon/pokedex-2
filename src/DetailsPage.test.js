import React from 'react';
import { shallow } from 'enzyme';
import DetailsPage from './DetailsPage.js';

it('renders single pokemon item details', () => {
    const wrapper = shallow(<DetailsPage character={{ pokemon: 'character',url_image: 'some image', type_1: 'some type', attack: 0, defense: 0}}/>);
  
    expect(wrapper).toMatchSnapshot();
  });