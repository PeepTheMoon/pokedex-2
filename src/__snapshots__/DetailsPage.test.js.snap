// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders single pokemon item details 1`] = `
<div>
  <PokemonItem
    character={
      Object {
        "pokemon": "",
      }
    }
  />
</div>
`;
